molgenis-data-annotators/src/main/java/org/molgenis/data/annotation/core/utils/AnnotatorUtils.java

```diff
@@ -63,14 +80,138 @@ else if (StringUtils.isEmpty(annotations))
 	/**
 	 * Adds a new compound attribute to an existing CrudRepository
 	 *
-	 * @param entityMetaData     {@link EntityMetaData} for the existing repository
-	 * @param attributeMetaDatas
+	 * @param entityMetaData           {@link EntityMetaData} for the existing repository
+	 * @param attributeMetaDataFactory
+	 * @param annotator
 	 */
 	public static EntityMetaData addAnnotatorMetadataToRepositories(EntityMetaData entityMetaData,
-			List<AttributeMetaData> attributeMetaDatas)
+			AttributeMetaDataFactory attributeMetaDataFactory, RepositoryAnnotator annotator)
 	{
-		attributeMetaDatas.stream().filter(part -> entityMetaData.getAttribute(part.getName()) == null)
-				.forEachOrdered(part -> entityMetaData.addAttribute(part));
+		List<AttributeMetaData> attributeMetaDatas = annotator.getOutputAttributes();
+		AttributeMetaData compound;
+		String compoundName = annotator.getFullName();
+		compound = entityMetaData.getAttribute(compoundName);
+		if (compound == null)
+		{
+			compound = attributeMetaDataFactory.create().setName(compoundName).setLabel(annotator.getFullName())
+					.setDataType(MolgenisFieldTypes.AttributeType.COMPOUND).setLabel(annotator.getSimpleName());
+			AttributeMetaData finalCompound = compound;
+			attributeMetaDatas.stream().filter(part -> entityMetaData.getAttribute(part.getName()) == null)
+					.forEachOrdered(part -> finalCompound.addAttributePart(part));
+			entityMetaData.addAttribute(compound);
+		}
 		return entityMetaData;
 	}
+
+	public static String annotate(RepositoryAnnotator annotator, VcfAttributes vcfAttributes,
```

Comment:
This is a very long method can you please cut it into smaller functions and add comments?